crypto/signer.go

```diff
@@ -24,30 +24,25 @@ import (
 )
 
 var (
-	signerHashLookup = map[sigpb.DigitallySigned_HashAlgorithm]crypto.Hash{
-		sigpb.DigitallySigned_SHA256: crypto.SHA256,
-	}
-	reverseSignerHashLookup = map[crypto.Hash]sigpb.DigitallySigned_HashAlgorithm{
+	sigpbHashLookup = map[crypto.Hash]sigpb.DigitallySigned_HashAlgorithm{
 		crypto.SHA256: sigpb.DigitallySigned_SHA256,
 	}
 )
 
 // Signer is responsible for signing log-related data and producing the appropriate
 // application specific signature objects.
 type Signer struct {
-	hash         crypto.Hash
-	signer       crypto.Signer
-	sigAlgorithm sigpb.DigitallySigned_SignatureAlgorithm
+	hash   crypto.Hash
+	signer PrivateKeyManager
 }
 
 // NewSigner creates a new Signer wrapping up a hasher and a signer. For the moment
 // we only support SHA256 hashing and either ECDSA or RSA signing but this is not enforced
 // here.
-func NewSigner(sigAlgo sigpb.DigitallySigned_SignatureAlgorithm, signer crypto.Signer) *Signer {
+func NewSigner(signer PrivateKeyManager) *Signer {
```

Comment:
So the question here is one of philosophy. 
Pro: This change makes it easier to create `Signer`s from `PrivateKeyManager`
Con: This now couples the Signer to KeyManager. Now I can't pass in my own generic private without invoking PrivateKeyManager.  
- This might make testing a bit harder since we'll need to create an extra object to create Signer.
- This means that this Signing library will be less useful to other projects that might not want to incorporate PrivateKeyManager.

What do you think?